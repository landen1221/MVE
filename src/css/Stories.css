.Stories p {
  margin-right: 4em;
}
