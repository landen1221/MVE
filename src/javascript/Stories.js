import { makeStyles } from "@material-ui/core/styles";
import Paper from "@material-ui/core/Paper";
import "../css/Stories.css";
import Story from "./Story";

const useStyles = makeStyles((theme) => ({
  root: {
    display: "flex",
    flexWrap: "wrap",
    "& > *": {
      margin: theme.spacing(1),
      width: theme.spacing(150),
      minHeight: theme.spacing(12),
    },
  },
}));

const Stories = ({ storyList }) => {
  const classes = useStyles();

  return (
    <div className={`Stories ${classes.root}`}>
      {storyList.length > 0 ? (
        storyList.map((currStory, idx) => (
          <Paper elevation={3} key={idx}>
            <Story currStory={currStory} />
          </Paper>
        ))
      ) : (
        <i id="no-stories">
          <br />
          no stories available
        </i>
      )}
    </div>
  );
};

export default Stories;
