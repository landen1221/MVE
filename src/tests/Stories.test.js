import React from "react";
import { render } from "@testing-library/react";
import Stories from "../javascript/Stories";
import { vaccines, storyList } from "./globalTestVariables";

// Props: storyList, search, vaccines
const search = "";

it("renders without crashing", () => {
  render(<Stories storyList={storyList} search={search} vaccines={vaccines} />);
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <Stories storyList={storyList} search={search} vaccines={vaccines} />
  );
  expect(asFragment()).toMatchSnapshot();
});
