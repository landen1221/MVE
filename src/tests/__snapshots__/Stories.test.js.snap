// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Stories makeStyles-root-2"
  >
    <i
      id="no-stories"
    >
      <br />
      no stories available
    </i>
  </div>
</DocumentFragment>
`;
